docs/source/trainer/checkpointing.rst

```diff
@@ -329,6 +329,34 @@ and delete the checkpoints from the local disk. The checkpoints will be located
 ``checkpoints/ep3.pt`` for third epoch's checkpoints, for example. The full URI in this case would be:
 ``s3://my_bucket/checkpoints/ep3.pt``.
 
+For uploading checkpoints to Coreweave's object store, the code is very similar to the
+above S3 uploading code. The only difference is you must set your Coreweave endpoint url.
+To do this you can just set the ``AWS_ENDPOINT_URL`` environment variable before creating the
+:class:`.Trainer`, like so:
+
+.. testcode::
+    :skipif: not _LIBCLOUD_INSTALLED
+
+    import os
+
+    os.environ['AWS_ENDPOINT_URL'] = 'https://object.las1.coreweave.com'
+    from composer.trainer import Trainer
+
+    # Save checkpoints every epoch to s3://my_bucket/checkpoints
+    trainer = Trainer(
+        model=model,
+        train_dataloader=train_dataloader,
+        max_duration='10ep',
+        save_folder='s3://my_bucket/checkpoints',
```

Comment:
Would it make more sense for the user to set a URI for coreweave (not sure what they use, we can ask them) That we use boto3/AWS to access coreweave seems like an implementation detail that the user should not need to be troubled with?

Reply:
yeah that's the weird thing is that they still use an s3 uri. See [here](https://docs.coreweave.com/storage/object-storage)